### PR TITLE
[keymgr,lint] Fix some width mismatches; explicitly slice some enum constants

### PR DESCRIPTION
**Pull request by rswarbrick**
Yet more minor tidy-ups in my quest to get TopEarlgrey lint-clean with Verilator :-)

### CODE REVIEW

File: hw/ip/keymgr/rtl/keymgr_reseed_ctrl.sv
Comment (tjaychen) [inline hw/ip/keymgr/rtl/keymgr_reseed_ctrl.sv:46]: this one will probably get a signed error from ascentlint also :)

Comment (rswarbrick) [reply hw/ip/keymgr/rtl/keymgr_reseed_ctrl.sv:46]: Gah, thanks. I'll redeclare it unsigned.

**Comment by rswarbrick** [comment]
@tjaychen: I've just added a follow-up commit to this PR (because the next batch of lint warnings come from the same code). I think this fixes an actual bug in the code: presumably we've never tested what comes out of this interface in "Adv", "Gen" or "Id" modes?

This sort of thing seems to be a strong argument for getting things lint-clean with Verilator too and/or strengthening the AscentLint rule set that we use.

**Comment by rswarbrick** [comment]
(Mostly a note to myself): This is blocked by the fact that we don't know how to slice some bits out of an enum without getting AscentLint warnings. See https://github.com/lowRISC/opentitan/pull/5904#discussion_r609044291.

**Comment by rswarbrick** [comment]
@tjaychen: I've switched to waiving the enum constant width warnings; I think this should be good to go?

**Review by tjaychen — APPROVED** [review]
lgtm! if it does trigger more ascentlint warnings for any reason, i can have a look.

**Pull request merged**
